python/selfie-lib/selfie_lib/CommentTracker.py

```diff
@@ -1,18 +1,46 @@
-from typing import Dict, Iterable, Tuple
+from typing import Dict, Iterable, Tuple, Sequence, TypeVar, Callable
+from abc import ABC, abstractmethod
 from enum import Enum, auto
 import threading
-from selfie_lib.TypedPath import TypedPath
+
 from selfie_lib.Slice import Slice
 
+from selfie_lib.TypedPath import TypedPath
+
+T = TypeVar("T")
+
+
+class FS(ABC):
+    @abstractmethod
+    def file_walk(self, typed_path, walk: Callable[[Sequence["TypedPath"]], T]) -> T:
+        pass
 
-# Placeholder implementations for CallStack, SnapshotFileLayout, and FS
-class CallStack:
-    pass
+    @abstractmethod
+    def file_read(self, typed_path) -> str:
+        return self.file_read_binary(typed_path).decode()
+
+    @abstractmethod
+    def file_write(self, typed_path, content: str):
+        self.file_write_binary(typed_path, content.encode())
+
+    @abstractmethod
+    def file_read_binary(self, typed_path) -> bytes:
+        pass
+
+    @abstractmethod
+    def file_write_binary(self, typed_path, content: bytes):
+        pass
+
+    @abstractmethod
+    def assert_failed(self, message: str, expected=None, actual=None) -> Exception:
+        pass
```

Comment:
This is fine for now, but just wanted to note that I have raised an issue for this

- https://github.com/diffplug/selfie-python-wip/issues/46